src/attached_probe.cpp

```diff
@@ -321,7 +321,11 @@ void AttachedProbe::load_prog()
 
   for (int attempt=0; attempt<3; attempt++)
   {
+#ifdef HAVE_BCC_PROG_LOAD
+    progfd_ = bcc_prog_load(progtype(probe_.type), namep,
+#else
```

Comment:
Maybe we could have something like this in a header file to avoid ifdefs when calling these functions:

```c++
#ifdef HAVE_BCC_PROG_LOAD
constexpr auto bcc_prog_load = bpf_prog_load;
#endif

#ifdef HAVE_BCC_CREATE_MAP
constexpr auto bcc_create_map = bpf_create_map;
#endif
```

Reply:
That's clever, but may make it harder to debug later: the function calls in the code may not be what they say they are.